graph/simple/simple.go

```diff
@@ -43,3 +45,37 @@ const maxInt int = int(^uint(0) >> 1)
 func isSame(a, b float64) bool {
 	return a == b || (math.IsNaN(a) && math.IsNaN(b))
 }
+
+// idset implements available ID storage.
+type idSet struct {
+	used, free intsets.Sparse
+}
+
+// newID returns a new unique ID.
+func (s *idSet) newID() int {
+	var id int
+	if s.free.Len() != 0 && s.free.TakeMin(&id) {
```

Comment:
I think this is residual from when the behaviour was map based. I've made the change, but it looks like we may end up back at the map-based approach anyway.